users/userRouter.js

```diff
@@ -33,14 +63,51 @@ router.put('/:id', (req, res) => {
 //custom middleware
 
 function validateUserId(req, res, next) {
-
+    const userId = req.params.userId;
+    const result = db.getById(userId)
+    result.then(response => {
+      if(res) {
+          // if id is valid store the user object as req.user 
+         req.user = response;
```

Comment:
You should probably rewrite this function a bit. Line 69 doesn't really make sense; you should be checking if you're getting a response from getByUserId, not checking the res. Also, the next() should be in that block, not it's own. You want the middleware function to call next if it passes the test you set.

Reply:
Notice that you're getting a response from db.getById, and you called it response. That's what you need to be checking. Inside that if block should be you setting the req.user to response and calling next. There should be one else after that that sends the error message.